src/lib/block_if.c

```diff
@@ -618,17 +618,17 @@ blockif_open(const char *optstr, const char *ident)
 	sectsz = DEV_BSIZE;
 	psectsz = psectoff = 0;
 	candelete = geom = 0;
+  blocks = 0;
 	if (S_ISCHR(sbuf.st_mode)) {
-		perror("xhyve: raw device support unimplemented");
-		goto err;
-		// if (ioctl(fd, DIOCGMEDIASIZE, &size) < 0 ||
-		// 	ioctl(fd, DIOCGSECTORSIZE, &sectsz))
-		// {
-		// 	perror("Could not fetch dev blk/sector size");
-		// 	goto err;
-		// }
-		// assert(size != 0);
-		// assert(sectsz != 0);
+		if (ioctl(fd, DKIOCGETBLOCKCOUNT, &blocks) < 0 ||
+        ioctl(fd, DKIOCGETBLOCKSIZE, &sectsz)) {
+		  perror("Could not fetch dev blk/sector size");
+		 	goto err;
+		}
+		assert(blocks != 0);
+		assert(sectsz != 0);
```

Comment:
Since they were commented out before I presume these originally come from bhyve and it would be better not to diverge on such trivial things. Maybe bhyve could be changed too though.

To that end, rebasing this PR onto #75 seems like a good idea to me.